docs: Adds riddle instructions for Slovenian devs (#404)
oasisprotocol/docs: docs/uganka.md

napišeš kratko skripto, ki se poveže s pametno pogodbo na našem omrežju,
tajno pridobi vprašanje in nanj odgovori. Če bo odgovor pravilen, osvojiš
kovančke ROSE! 🌹

## Blockchain, Ethereum, Solidity, Dapp, kaj za vraga?

Poznaš le C? Python? Javascript? Odlično! Vsega ostalega se boš priučil. 🎓

Porazdeljena omrežja niso enostavna reč. Programiranje porazdeljenih
aplikacij ima veliko omejitev v primerjavi z namiznimi ali spletnimi,
zato je potrebno imeti odprto glavo. Poznavanje algoritmov
in podatkovnih struktur, pisanje programske kode, ki učinkovito reši problem,
iznajdljivost pri iskanju hroščev, natančno sledenje poteku programa z
vidika varnosti in kriptografske rešitve so cenjene vrline.

## OK. Kaj moram storiti?

1. Najprej se nam pridruži na kanalu `#slovenia` na
   [Oasisovem Discord strežniku][oasis-discord]. Do kanala pridobiš dostop
   tako, da v kanalu `#get-your-role` klikneš na slovensko zastavico 🇸🇮.
   Če želiš, se lahko na kratko predstaviš. Lahko pa napišeš le `živjo` in
   bomo zadovoljni. Če naletiš na nepremostljivo oviro pri reševanju
   uganke, ti priskočimo na pomoč. Na neprimerna vprašanja ti bomo odgovorili
   s še bolj neprimernim odgovorom. 😉

> 2. Če si nov v blokovnih verigah in bo to tvoja prva porazdeljena aplikacija,

Reviewer: ```suggestion
2. Če so ti blokovne verige nove in bo to tvoja prva porazdeljena aplikacija,
```

   preglej [kratek Ethereumov vodič po pametnih pogodbah][ethereum-tutorial]. 💡

> 3. Bravo! Spoštovanega mojstra v programiranju pametnih pogodb za Ethereum
>    zdaj naprošamo, da si prebere, kako bi sprogramiral *tajno* pametno pogodbo

Reviewer: ```suggestion
3. Bravo! Sedaj, ko obvladaš programiranje pametnih pogodb za Ethereum, poglej še *tajne* pametne pogodbe
```
Tale poved mi je kar dolga, uporabi, če se ti zdi koristno

Author: Sem odvisnike zamenjal s prilastki in je krajše ;)

        require(bytes(_coupons[coupon]).length != 0, errInvalidCoupon);
        _numLookups++;

        return _questions[computeQuestionIndex(coupon)];
    }

    // Check the answer to the question for given coupon.
    function submitAnswer(string memory coupon, string memory answer) external returns (string memory) {
        require(bytes(_coupons[coupon]).length != 0, errInvalidCoupon);
        _numAnswers++;
        require(keccak256(bytes(_answers[computeQuestionIndex(coupon)])) == keccak256(bytes(answer)), errWrongAnswer);
        _numCorrectAnswers++;

        return string.concat("Congratulations, your answer is correct! Use this private key to access reward on Oasis Sapphire Mainnet: ", _coupons[coupon]);
    }

    // Compute the index of the question for given coupon.
    function computeQuestionIndex(string memory coupon) private view returns (uint) {
        return uint(keccak256(bytes(coupon))) % _questions.length;
    }

    // Return statistics.
    // NB: This can include duplicate queries and answers!
    function getStatistics() external view returns (uint, uint, uint) {
        return (_numLookups, _numAnswers, _numCorrectAnswers);
    }
}
```

> 5. Tvoja naloga je, da s pomočjo orodij, ki si jih doslej spoznal, napišeš

Reviewer: ```suggestion
5. Tvoja naloga je, da s pomočjo orodij, ki jih sedaj poznaš, napišeš
```

   skripto, ki komunicira s pametno pogodbo uganke. Najprej s *tajno
   transakcijo* do `getQuestion()` poizvedi za vprašanjem, vezanem na tvoj
   kupon z letaka. Nato sestavi odgovor in ga s tajno transakcijo do
   `submitAnswer()` pošlji nazaj. Če bo odgovor pravilen, se ti bo razkril
   zasebni ključ do čisto pravcatih kovančkov ROSE na verigi
   [Oasis Sapphire Mainnet][oasis-explorer]. 🎉

6. Čestitamo! Kovančke lahko uporabiš za objavo novih porazdeljenih

>    aplikacij na omrežju Oasis in prvi na svetu izkoristiš *tajnost*

Reviewer: ```suggestion
   aplikacij na omrežju Oasis in med prvimi na svetu izkoristiš *tajnost*
```
